repl/src/main/scala/ammonite/repl/frontend/JLineFrontend.scala

```diff
@@ -107,7 +107,8 @@ object JLineFrontend{
 
       case Res.Success(ev) =>
         val last = reader.getHistory.size()-1
-        reader.getHistory.set(last, buffered + reader.getHistory.get(last))
+        if (last >= 0)
```

Comment:
The problem is that this code is run on the first predefined command, which leads to an exception if the history is initially empty. Maybe a better approach would be to inhibit calling `frontend.update` when running the predefined commands? BTW, what is your take on adding the predefined commands to the history?